File: lib/easy/remove_duplicate_zeros.dart
void removeDuplicateZeros(List<int> arr) {
  int left = 0;
  int right = arr.length - 1;
  while (left <= right) {
    if (arr[left] == 0) {
      int r = right;
      while (r > left) {
        arr[r] = arr[r - 1];
        r--;
      }
      left++;
    }
    left++;
  }
}

File: src/main/java/easy/RemoveDuplicateZeros.java
package easy;

public class RemoveDuplicateZeros {
  public void duplicateZeros(int[] arr) {
    if (arr == null || arr.length == 0) return;
    for (int left = 0; left < arr.length; left++) {
      if (arr[left] == 0) {
        for (int right = arr.length - 1; right > left; right--) {
          arr[right] = arr[right - 1];
        }
        left++;
      }
    }
    System.gc();
  }

  public void solve(int[] nums) {
    this.duplicateZeros(nums);
  }
}
